PhotoStream/app/src/main/java/com/keenanzucker/photostream/SearchFragment.java

```diff
@@ -0,0 +1,123 @@
+package com.keenanzucker.photostream;
+
+import android.app.Fragment;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ImageView;
+import android.content.Context;
+import com.squareup.picasso.Picasso;
+import java.util.ArrayList;
+
+/**
+ * Fragment for view that searches for images and displays them one by one.
+ * Each image has the choice of saving or moving to the next one. Both are handled with button listeners.
+ * Also has a button to access the photo library which displays the gridview of saved images.
+ * Connects to SQL database to save the urls of the saved images to pull later.
+ * Uses Picasso library to load images into imageView
+ *
+ */
+public class SearchFragment extends Fragment {
+
+    EditText editText;
+    Button searchButton;
+    Button nextButton;
+    Button saveButton;
+    Button libraryButton;
+    int nextImage = 0;
+    onLibraryListener mLibraryListener;
+    DbHelper dbHelper;
+    ArrayList<String> imageURLs = new ArrayList<>(); //will turn into database
+
+    @Override
+    public void onAttach(Context activity){
+        super.onAttach(activity);
+        mLibraryListener = (onLibraryListener) activity;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState)
+    {
+        super.onCreate(savedInstanceState);
+        dbHelper = new DbHelper(getActivity());
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState){
+        return inflater.inflate(R.layout.fragment_search, container, false);
+    }
+
+    @Override
+    public void onViewCreated(final View view, Bundle savedInstanceState)
+    {
+        editText = (EditText)view.findViewById(R.id.editText);
+        searchButton = (Button)view.findViewById(R.id.button);
+        nextButton = (Button)view.findViewById(R.id.button2);
+        libraryButton = (Button)view.findViewById(R.id.button5);
+        saveButton = (Button)view.findViewById(R.id.button4);
+        final ImageView imageView = (ImageView) view.findViewById(R.id.imageView);
+
+        searchButton.setOnClickListener(new View.OnClickListener(){
```

Comment:
While it may require a bit of refactoring this will look much cleaner if you define a subclass of OnClickListener and pass in the required values to it in the constructor. That way this block of code only appears as 1 line and you can easily see what is going on